ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/InMemoryCUQuotaSource.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota.capacityunit;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.AccountService;
+import com.github.ambry.config.QuotaConfig;
+import com.github.ambry.quota.Quota;
+import com.github.ambry.quota.QuotaException;
+import com.github.ambry.quota.QuotaName;
+import com.github.ambry.quota.QuotaResource;
+import com.github.ambry.quota.QuotaResourceType;
+import com.github.ambry.quota.QuotaSource;
+import com.github.ambry.quota.QuotaUtils;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A {@link QuotaSource} implementation that keeps the quota and usage values in memory, and treats Ambry frontend's
+ * Read and Write Capacity Units as system resources for handling quota exceeded requests.
+ */
+public class InMemoryCUQuotaSource implements QuotaSource {
```

Comment:
I am not sure if InMemoryCUQuotaSource is a good name. Usage is calculated locally, in other words, it will always be stateless. I cannot think of any other implementation other than "InMemory". I suggest to remove the InMemory part